firmware/src/src.ino

```diff
@@ -222,11 +250,37 @@ void sensorLoop(){
 }
 
 // #region helpers
-// C is disgusting and I hate it deeply...
+//-----------------------------------------------------------------------------
+void send_warning(String topic, String msg) {
+  String clean_msg = msg;
+  clean_msg.replace(',', '_');
+  clean_msg.replace("\n", "");
+  String warn = "2,Arduino,1,"; 
+  warn += topic;
+  warn += " ";
+  warn += msg.length();
+  warn += " bytes: |"; 
+  warn += clean_msg;
+  warn += "|\n";
+  Serial.print(warn);
+  Serial.flush();
+}
+
+//-----------------------------------------------------------------------------
+// Resets our global string and flag.
+void resetMessage() {
+  for(unsigned i=0; i < MESSAGE_LENGTH; i++){
+    message.setCharAt(i, '\0');
+  }
+  message = "";
+  stringComplete = false;
+}
+
+//-----------------------------------------------------------------------------
 int split(String messages, String* splitMessages,  char delimiter){
   int indexOfComma = 0;
   int delim_count = 0;
-  for(int i = 0; messages.indexOf(delimiter, indexOfComma) > 0; i++){
```

Comment:
If you get a too long message, you will write out of bounds in splitMessages and stomp on memory!  So only read up to a maximum limit (of the size of splitMessages).